dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -264,29 +260,151 @@ def central_frequency(self):
+
+        return trigger_type
+
+
+    @property
+    def single_threshold_trigger(self):
+        logger.info(self.trigger_type)
+        if self.trigger_type != 'single_threshold':
+            return False
+        threshold = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)['threshold']
+        return {'threshold' : threshold}
+
+
+    def set_single_threshold_trigger(self, threshold):
+        payload = {'threshold' : threshold, 'threshold_high' : 0.0, 'n_triggers' : 1, 'channel' : self.channel_id}
+        result = self.provider.cmd(self.psyllid_interface, 'set_trigger_configuration', payload = payload)
+
+        if result['values'][0] == False:
+            logger.error('Could not apply trigger settings')
+            raise core.exceptions.DriplineGenericDAQError('Could not apply trigger settings. Psyllid instance not in triggered mode')
+        else:
+            result = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)
+            return {'threshold' : result['threshold']}
+
+
+    @property
+    def two_threshold_trigger(self):
+        if self.trigger_type != 'two_thresholds':
+            return False
+        result = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)
+        return {'low_threshold' : result['threshold'], 'high_threshold' : result['threshold_high']}
+
+
+    def set_two_threshold_trigger(self, high_threshold, low_threshold):
+        payload = {'threshold' : low_threshold, 'threshold_high' : high_threshold, 'n_triggers' : 1, 'channel' : self.channel_id}
+        result = self.provider.cmd(self.psyllid_interface, 'set_trigger_configuration', payload = payload)
+
+        if result['values'][0] == False:
+            logger.error('Could not apply trigger settins') 
+            raise core.exceptions.DriplineGenericDAQError('Could not apply trigger settings. Psyllid instance not in triggered mode')
+        else:
+            result = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)
+            return {'low_threshold' : result['threshold'], 'high_threshold' : result['threshold_high']}
+
+
+    @property
+    def multi_trigger(self):
+        if self.trigger_type != 'multi_trigger':
+            return False
+        result = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)
+        skip_tolerance = self.provider.cmd(self.psyllid_interface, 'get_time_window', payload = self.payload_channel)['skip_tolerance']
+
+        # In addition to threshold and n_triggers, returns skip_tolerance as it is n_triggers/skip_tolerance that defines this trigger
+        return {'threshold' : result['threshold'], 'n_triggers' : result['n_triggers'], 'skip_tolerance': skip_tolerance}
+
+
+    def set_multi_trigger(self, threshold, n_triggers):
```

Comment:
if multi_trigger is an `@property` then probably `set_multi_trigger` should actually be:
```
@multi_trigger.setter
def multi_trigger(self, value):
    threshold, n_triggers = value
    ...
```

Reply:
I totally agree but don't know how to make an attribute setter that takes more than 1 argument.
Can you tell me how to do that?

Reply:
Oh, you'd have to either pack the pair of values into a tuple that could be unpacked (that is, if `value == (threshold_val, n_triggers_val)`.... I think that that should be doable from code, but the CLI for dragonfly probably doesn't support it.

The way drip line would prefer it done would be to split them into multiple properties. One for the threshold and another for the number of triggers, each with some default value. That probably requires some refactoring so I'll let it go unless you think that sounds easy.

Reply:
I've had pretty spotty luck with CLI parsing of lists and things.  If we need this we could do some testing of dripline_agent to get the desired behavior.